keras_cv/layers/preprocessing/random_flip_test.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.random_flip import RandomFlip
+
+
+class RandomFlipTest(tf.test.TestCase):
+    def _run_test(self, mode, expected_output=None, mock_random=None):
+        np.random.seed(1337)
+        num_samples = 2
+        orig_height = 5
+        orig_width = 8
+        channels = 3
+        if mock_random is None:
+            mock_random = [0.6 for _ in range(num_samples)]
+            if mode == "horizontal_and_vertical":
+                mock_random *= 2
+        inp = np.random.random((num_samples, orig_height, orig_width, channels))
+        if expected_output is None:
+            expected_output = inp
+            if mode == "horizontal" or mode == "horizontal_and_vertical":
+                expected_output = np.flip(expected_output, axis=2)
+            if mode == "vertical" or mode == "horizontal_and_vertical":
+                expected_output = np.flip(expected_output, axis=1)
+        layer = RandomFlip(mode)
+        with unittest.mock.patch.object(
+            layer._random_generator,
+            "random_uniform",
+            side_effect=mock_random,
+        ):
+            actual_output = layer(inp, training=True)
+            self.assertAllClose(expected_output, actual_output)
+
+    def test_random_flip(self):
```

Comment:
can we use absl.testing.parameterized to handle these cases?  An example of  how to use parameterized is here: https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/serialization_test.py

Reply:
done!